diff --git a/meson.build b/meson.build
--- a/meson.build
+++ b/meson.build
@@ -508,18 +508,26 @@
 
 features += {'win32-desktop': win32 and not uwp.found()}
 if features['win32-desktop']
-    win32_desktop_libs = [cc.find_library('avrt'),
-                          cc.find_library('dwmapi'),
-                          cc.find_library('gdi32'),
-                          cc.find_library('imm32'),
-                          cc.find_library('ntdll'),
-                          cc.find_library('ole32'),
-                          cc.find_library('pathcch'),
-                          cc.find_library('shcore'),
-                          cc.find_library('user32'),
-                          cc.find_library('uuid'),
-                          cc.find_library('uxtheme'),
-                          cc.find_library('version')]
+    no_static_win = meson.get_external_property('no_static_windows_libs', false)
+    if no_static_win
+        win32_desktop_libs = [declare_dependency(link_args: [
+            '-lavrt', '-ldwmapi', '-lgdi32', '-limm32', '-lntdll', '-lole32',
+            '-lpathcch', '-lshcore', '-luser32', '-luuid', '-luxtheme', '-lversion',
+        ])]
+    else
+        win32_desktop_libs = [cc.find_library('avrt'),
+                              cc.find_library('dwmapi'),
+                              cc.find_library('gdi32'),
+                              cc.find_library('imm32'),
+                              cc.find_library('ntdll'),
+                              cc.find_library('ole32'),
+                              cc.find_library('pathcch'),
+                              cc.find_library('shcore'),
+                              cc.find_library('user32'),
+                              cc.find_library('uuid'),
+                              cc.find_library('uxtheme'),
+                              cc.find_library('version')]
+    endif
     dependencies += win32_desktop_libs
     path_source = files('osdep/path-win.c')
     if not get_option('fuzzers')
