diff --git a/meson.build b/meson.build
--- a/meson.build
+++ b/meson.build
@@ -1783,7 +1783,11 @@ features += {'libmpv-' + get_option('default_library'): get_option('libmpv')}
 # build targets
 if win32
     windows = import('windows')
-    res_flags = ['--codepage=65001']
+    if cc.get_id() in ['msvc', 'clang-cl']
+        res_flags = []
+    else
+        res_flags = ['--codepage=65001']
+    endif
 
     # Unintuitively, this compile operates out of the osdep subdirectory.
     # Hence, these includes are needed.
